provider/core/src/data_provider.rs

```diff
@@ -947,22 +951,26 @@ fn test_debug() {
 /// - [`HelloWorldProvider`](crate::hello_world::HelloWorldProvider)
 /// - [`AnyPayloadProvider`](crate::struct_provider::AnyPayloadProvider)
 /// - [`InvariantDataProvider`](crate::inv::InvariantDataProvider)
-pub trait DataProvider<M>
+pub trait DynProvider<M>
```

Comment:
thought: the name DynProvider feels weird to me. unsure what it should be, but it's not ... really producing dyns?

Reply:
I came up with `DynProvider` because (1) I wanted to change it away from `DataProvider` and (2) its main purpose moving forward is to produce `SerializeMarker`, `AnyMarker`, `CrabBakeMarker`, etc., which are all dyns.

Reply:
Ah, perhaps we should document it as such that it's largely intended to be implemented in ways that produce dyn markers?

Reply:
I added better docs.  Please re-open this thread if the new docs are not good.